### PR TITLE
Script to download the source based on tag

### PR DESCRIPTION
**Pull request by PankajWorks**

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
```
Comment (yhemanth) [inline install.sh]: Must be Apache licensed. This is freely available.

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+SOURCE_TAR_NAME=dp-cluster-reg.tar
+TARGET_FOLDER=dp-cluster-reg
+if command -v python &>/dev/null
+then
+    echo "Python  is installed"
+else
+    echo "Python is not installed"
+fi
+
+if [ -z ${RELEASE+x} ] || [ -z "$RELEASE" ]
```
Comment (yhemanth) [inline install.sh]: Not sure what the ${RELEASE+x} does - can you clarify?

File: install.sh [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    tar -xvzf ${SOURCE_TAR_NAME} -C ${TARGET_FOLDER} --strip-components 1
+    UNTAR_STATUS=$?
+fi
+
+
+if [ "x$UNTAR_STATUS" == "x0" ]
+then
+    echo "Successfully extracted the script in ${TARGET_FOLDER}"
+    echo "You can now run"
+    echo "cd ${TARGET_FOLDER};python dp-cluster-setup-utility.py"
```
Comment (yhemanth) [inline install.sh]: Can we also automate install of the CM client dependency? It would help greatly.

File: install.sh (hunk 2 above)
Comment (PankajWorks) [reply install.sh]: It checks : if a variable is specifically unset (as distinct from an empty string)

File: install.sh [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+fi
+
+if [ "$CM" = true ]
+then
+    if [ "$VIRTUAL_ENV" != "" ]
+    then
+    echo "Installing cm_client in virtualenv $VIRTUAL_ENV"
+    elif [ "$VIRTUAL_ENV" == "" ]
+    then
+        echo "Installing cm_client in globl scope"
```
Comment (yhemanth) [inline install.sh]: typo: global

File: install.sh [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+else
+    echo "Python is not installed. Install Python an proceed with installation"
+    exit 1
+fi
+
+if [ "$CM" = true ]
+then
+    if [ "$VIRTUAL_ENV" != "" ]
+    then
+    echo "Installing cm_client in virtualenv $VIRTUAL_ENV"
```
Comment (yhemanth) [inline install.sh]: Indentation please.

File: install.sh [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    elif [ "$VIRTUAL_ENV" == "" ]
+    then
+        echo "Installing cm_client in globl scope"
+    fi
+    pip install cm_client
```
Comment (yhemanth) [inline install.sh]: This is the same whether VIRTUAL_ENV is set or not, so how does this matter. If it is just for the print, I don't know if that's so interesting.

File: install.sh [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    status=$?
+    if [ $status -eq 0 ]
+    then
+        echo "Installation successful"
+    else
+        echo "Installation failed. resolve installation isuue and re-run the script"
```
Comment (yhemanth) [inline install.sh]: Typo: please change the message to: "Installation of cm_client module required for running the cluster registration script failed. Please install this manually and re-run the script"

File: install.sh (hunk 6 above)
Comment (PankajWorks) [reply install.sh]: Its a information for user that where the module will be installed.

**Pull request merged**